### PR TITLE
Enhanced link data type discovery

### PR DESCRIPTION
**Pull request by acouch**
https://github.com/NuCivic/dkan/issues/739

This PR makes the following enhancements:
- Replaces `remote_file_source` module remote file link with a native one
- Adds `getRemoteFileInfo.php` library for better file type and file name discovery

This allows us to us the remote file link for files with URLs like `https://data.example.gov/api/views/abc-123/rows.csv?accessType=DOWNLOAD` or  `http://demo.getdkan.com/node/5/download` which are common. It also implements proper filemime discovery for linked URLs.

TODO:
- [ ] Further prepare QA
- [ ] Add more unit tests
## Acceptance Criteria

**Add resource with 'Link to a file'**
- https://data.wa.gov/api/views/mu24-67ke/rows.csv?accessType=DOWNLOAD
- http://demo.getdkan.com/node/5/download
- http://api.tiles.mapbox.com/v3/tmcw.map-gdv4cswo/markers.geojson
- http://gisdata.nd.gov/Metadata/ISO/counties.xml

**Add resource with 'Link to an api'**
- http://api.tiles.mapbox.com/v3/tmcw.map-gdv4cswo/markers.geojson
- http://gisdata.nd.gov/Metadata/Images/counties.png
- https://health.data.ny.gov/api/file_data/QLQGgv7WxWH8JDrpbySPD-y7pk4AYw9957WUVl1umwM?filename=NYSDOH_NHCostReport_2011.zip
## PR dependency

https://github.com/NuCivic/dkan/pull/1306

### CODE REVIEW

**Comment by nucivicmachine** [comment]
QA site created: http://enhanced_link_data_type_discovery.dkan_dataset.qa.nuamsdev.com

**Comment by nucivicmachine** [comment]
QA site created: http://enhanced_link_data_type_discovery.dkan_dataset.qa.nuamsdev.com

**Comment by nucivicmachine** [comment]
QA site created: http://enhanced_link_data_type_discovery.dkan_dataset.qa.nuamsdev.com

**Comment by nucivicmachine** [comment]
QA site created: http://enhanced_link_data_type_discovery.dkan_dataset.qa.nuamsdev.com

File: modules/dkan_dataset_content_types/dkan_dataset_content_types.features.field_instance.inc [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ uk-ogl|UK Open Government Licence (OGL)',
             'clipboard' => 0,
             'reference' => 0,
             'remote' => 0,
-            'remotefile' => 'remotefile',
+            'dkan_remotefile' => 'dkan_remotefile',
```
Comment (teosibileau) [inline modules/dkan_dataset_content_types/dkan_dataset_content_types.features.field_instance.inc:1587]: Potential for disaster here if feature gets updated (i think)

**Comment by dafeder** [comment]
Superceded by #276

**Pull request closed without merging**